app/Models/Movie.php

```diff
@@ -34,6 +34,7 @@ public function addTrailer(string $string_url) {
     }
     public function addRating(User $user, int $rating) {
         if ($user instanceof User && is_int($rating)) {
```

Comment:
Should use `if ($user != NULL && is_int($rating))` Instead, as the type constrain in the function argument list already forbids any other type.